src/dmd/mars.d

```diff
@@ -1738,12 +1746,19 @@ private bool parseCommandLine(const ref Strings arguments, const size_t argc, re
                             import dmd.cli : Usage;
                             string buf = `case "all":`;
                             foreach (t; Usage.transitions)
-                                buf ~= `params.`~t.paramName~` = true;`;
+                                if (!t.deprecated_)
+                                    buf ~= `params.`~t.paramName~` = true;`;
                             buf ~= "break;";
 
                             foreach (t; Usage.transitions)
                             {
-                                buf ~= `case "`~t.name~`": params.`~t.paramName~` = true;break;`;
+                                buf ~= `case "`~t.name~`":`;
+                                if (t.deprecated_)
+                                    buf ~= `fprintf(global.stdmsg, "-transition=`~t.name~` has no effect anymore\n");`;
```

Comment:
We can't trigger a real deprecation error because Phobos uses `-transition=complex` for now.